src/ui/profile_preferences.ui

```diff
@@ -707,7 +704,19 @@
             </property>
            </widget>
           </item>
-          <item row="2" column="0" colspan="4">
+          <item row="2" column="0" colspan="3">
+           <widget class="QLabel" name="label_variableWidthFontWarning">
+            <property name="font">
+             <font>
+              <pointsize>8</pointsize>
+             </font>
+            </property>
+            <property name="text">
+             <string>This font isn't monospace, which is not ideal for playing text games: you're welcome to use it but beware of issues</string>
```

Comment:
Not sure that this precise text is helpful - remember that CJK users are automatically NOT going to be using mono-space fonts (they are probably relying on fall-back behaviour where Qt/font-config is pulling in glyphs from a font that provides the ideographic graphemes they need). How about:
```
Some MUD games expect their player to be using a console type client which uses a mono-spaced font, this font is not one of those; you are welcome to use it but things like diagrams and in-game maps built up from rectangles of characters may not render correctly.
```

I do have medium to long term plans to revamp the `TTextEdit` class to measure the width of *every* glyph that is painted on screen and when the width of that glyph divided by the *normal* character width that `widechat_width.h` says it should have (i.e. `1` or `2`) ***exceeds*** the width that has been used (which most recently is `QFontMetrics(<font>).averageCharWidth()`) then the width gets re-adjusted to match the width needed for THAT glyph and then everything get repainted with each glyph centre-aligned in the new wider character cell width.  Effectively this should make a mono (or rather duo)-spaced font out of *any* font - whilst this might look a bit odd when used on proportional fonts it should not look *too* weird except on highly [cursive/script](https://en.wikipedia.org/wiki/Cursive) type fonts where the joins between the glyph will not *join-up*...

Reply:
> TTextEdit class to measure the width of every glyph that is painted on screen and when the width of that glyph divided by the normal character width that widechat_width.h says it should have (i.e. 1 or 2) exceeds the width that has been used (which most recently is QFontMetrics(<font>).averageCharWidth()) then the width gets re-adjusted to match the width needed for THAT glyph and then everything get repainted with each glyph centre-aligned in the new wider character cell width.

Wouldn't that be a lot slower than what we're doing now? What problem would it solve?

Reply:
It would make *any* font line up in the manner that a mono-spaced one does (it would also prevent the clipping that can occur with coloured emoji glyphs - as soon as one occurs the spacing would be increased everywhere in that `TTextEdit` to accommodate it).  As for being slow I am not sure that it would be that bad after an initial process/scan through the text that establishes just how width each character "*cell*" needs to be - the slow down would be due to repaints as a new (previously unseen) wider than previously encountered glyph is found.

Anyhow this is just an idea that I would like to test when I (or perhaps someone else) gets the time...

Reply:
Well, we can measure the performance of that when it's ready.

> CJK users are automatically NOT going to be using mono-space fonts

That's not correct - I asked one and they do use monospace fonts, compare the two screenshots:

https://cdn.discordapp.com/attachments/601798233332776960/682434816196476945/unknown.png

https://cdn.discordapp.com/attachments/601798233332776960/682434870722691162/unknown.png

> How about:

That text is too long for UI :( I think the one I have more or less gets the point across.

Reply:
But *neither* of those are (solely) using a mono-spaced font... :confounded:

Reply:
Strictly speaking no, but as far as the usual player is concerned, the second one is monospace as they told me :) it's ok to not be perfectly accurate because it'll get the point across better.

Reply:
> "...you're welcome to use it..."

That is rather condescending IMHO - it is sort of saying "we, the Mudlet makers, think you would be stupid/misguided/out-of-your-mind to choose it but we don't think you should" - so how about:
```
This font is not monospace, which may not be ideal for playing some text games: you can use it but there could be issues with aligning columns of text.
```
This gives a hint as to the main problem as to why you might not want to choose the particular font but does not rule it out as a silly idea.

Reply:
Thanks for spotting that, changed it.